.github/workflows/ci.yml

```diff
@@ -0,0 +1,47 @@
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+  push: # Run CI on the main branch after every merge. This is important to fill the GitHub Actions cache in a way that pull requests can see it
+    branches:
+      - main
+
+jobs:
+  ci:
+    name: ci
+    runs-on: ubuntu-20.04
+    timeout-minutes: 90
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly-2022-10-18
+          components: rustfmt, clippy
+          target: wasm32-unknown-unknown
+          override: true
+          default: true
+
+      - name: Set up cargo cache
+        uses: actions/cache@v3
+        continue-on-error: false
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Check Code
+        run: cargo fmt --all --check --manifest-path ./pallets/parachain-staking/Cargo.toml
+
+      - name: Test Code
+        run: cargo test --manifest-path ./pallets/parachain-staking/Cargo.toml
+
+      - name: Build
+        run: cargo build --locked --release --manifest-path ./pallets/parachain-staking/Cargo.toml
```

Comment:
We will add one more test `cargo test --features runtime-benchmarks` but this will not pass at the moment so we could leave it commented out for now?

Reply:
added 👍